examples/multigpu/graphbolt/node_classification.py

```diff
@@ -0,0 +1,444 @@
+"""
+This script trains and tests a GraphSAGE model for node classification on
+multiple GPUs using distributed data-parallel training (DDP) and GraphBolt
+data loader. 
+
+Before reading this example, please familiar yourself with graphsage node
+classification using GtaphBolt data loader by reading the example in the
+`examples/sampling/graphbolt/node_classification.py`.
+
+For the usage of DDP provided by PyTorch, please read its documentation:
+https://pytorch.org/tutorials/beginner/dist_overview.html and
+https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParal
+lel.html
+
+This flowchart describes the main functional sequence of the provided example:
+main
+│
+├───> OnDiskDataset pre-processing
+│
+└───> run (multiprocessing) 
+      │
+      ├───> Init process group and build distributed SAGE model (HIGHLIGHT)
+      │
+      ├───> train
+      │     │
+      │     ├───> Get GraphBolt dataloader with DistributedItemSampler
+      │     │     (HIGHLIGHT)
+      │     │
+      │     └───> Training loop
+      │           │
+      │           ├───> SAGE.forward
+      │           │
+      │           ├───> Validation set evaluation
+      │           │
+      │           └───> Collect accuracy and loss from all ranks (HIGHLIGHT)
+      │
+      └───> Test set evaluation
+"""
+import argparse
+import os
+
+import dgl.graphbolt as gb
+import dgl.nn as dglnn
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.nn as nn
+import torch.nn.functional as F
+import torchmetrics.functional as MF
+import tqdm
+from torch.distributed.algorithms.join import Join
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_size, hidden_size, out_size):
+        super().__init__()
+        self.layers = nn.ModuleList()
+        # Three-layer GraphSAGE-mean.
+        self.layers.append(dglnn.SAGEConv(in_size, hidden_size, "mean"))
+        self.layers.append(dglnn.SAGEConv(hidden_size, hidden_size, "mean"))
+        self.layers.append(dglnn.SAGEConv(hidden_size, out_size, "mean"))
+        self.dropout = nn.Dropout(0.5)
+        self.hidden_size = hidden_size
+        self.out_size = out_size
+        # Set the dtype for the layers manually.
+        self.set_layer_dtype(torch.float64)
```

Comment:
Why do we set the layer dtype to double here, while the model is cast to float32 below?